### DISCUSSION TITLE
snmpv3 - authProtocl: SHA-256, privProtocol: AES256

### DISCUSSION THREAD

**Discussion by isrijaya** [splunk/splunk-connect-for-snmp#970] [General]
Hello,

We have a trap source which uses SHA-256 and AES256 as the authentication and privacy protocol respectively.

How do I code those protocols in the secret?

I was trying the following:

microk8s kubectl create -n sc4snmp secret generic snmpv3-secret-sha2-aes \
  --from-literal=userName=USER02 \
  --from-literal=authKey=v0dr@aD! \
  --from-literal=privKey=v0dr@aD! \
  --from-literal=authProtocol=SHA-2 \
  --from-literal=privProtocol=AES256

and It gives error in the trap pods:

> 2024-03-04 01:28:59,881 DEBUG authProtocol: SHA-2
> 2024-03-04 01:28:59,882 DEBUG privProtocol: AES256
> Traceback (most recent call last):
>   File "/app/.venv/lib/python3.10/site-packages/pyasn1/type/univ.py", line 1209, in prettyIn
>     return tuple([int(subOid) for subOid in value.split(".") if subOid])
>   File "/app/.venv/lib/python3.10/site-packages/pyasn1/type/univ.py", line 1209, in <listcomp>
>     return tuple([int(subOid) for subOid in value.split(".") if subOid])
> ValueError**: invalid literal** for int() with base 10: 'NONE'
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/app/.venv/lib/python3.10/site-packages/pysnmp/smi/mibs/SNMPv2-SMI.py", line 659, in setValue
>     return self.syntax.setValue(value)
>   File "/app/.venv/lib/python3.10/site-packages/pysnmp/smi/mibs/SNMPv2-TC.py", line 51, in setValue
>     return self.clone(value)
>   File "/app/.venv/lib/python3.10/site-packages/pyasn1/type/base.py", line 391, in clone
>     return self.__class__(value, **initializers)
>   File "/app/.venv/lib/python3.10/site-packages/pyasn1/type/base.py", line 284, in __init__
>     value = self.prettyIn(value)
>   File "/app/.venv/lib/python3.10/site-packages/pysnmp/smi/mibs/SNMPv2-TC.py", line 370, in prettyIn
>     return base.prettyIn(self, value)
>   File "/app/.venv/lib/python3.10/site-packages/pyasn1/type/univ.py", line 1211, in prettyIn
>     raise error.PyAsn1Error(
> pyasn1.error.PyAsn1Error: Malformed Object ID NONE at AutonomousType: invalid literal for int() with base 10: 'NONE'
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/app/.venv/lib/python3.10/site-packages/pysnmp/smi/mibs/SNMPv2-SMI.py", line 732, in writeTest
>     self.__newSyntax = self.setValue(val, name, idx)
>   File "/app/.venv/lib/python3.10/site-packages/pysnmp/smi/mibs/SNMPv2-SMI.py", line 669, in setValue
>     raise error.WrongValueError(idx=idx, name=name, msg=exc_v)
> pysnmp.smi.error.WrongValueError: WrongValueError({'idx': 3, 'name': (1, 3, 6, 1, 6, 3, 15, 1, 2, 2, 1, 5, 5, 128, 0, 58, 140, 4, 6, 85, 83, 69, 82, 48, 50), 'msg': PyAsn1Error("Malformed Object ID NONE at AutonomousType: invalid literal for int() with base 10: 'NONE'")})
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/app/.venv/bin/traps", line 8, in <module>
>     sys.exit(main())
>   File "/app/.venv/lib/python3.10/site-packages/splunk_connect_for_snmp/traps.py", line 153, in main
>     config.addV3User(
>   File "/app/.venv/lib/python3.10/site-packages/pysnmp/entity/config.py", line 181, in addV3User
>     snmpEngine.msgAndPduDsp.mibInstrumController.writeVars(
>   File "/app/.venv/lib/python3.10/site-packages/pysnmp/smi/instrum.py", line 253, in writeVars
>     return self.flipFlopFsm(self.fsmWriteVar, varBinds, acInfo)
>   File "/app/.venv/lib/python3.10/site-packages/pysnmp/smi/instrum.py", line 239, in flipFlopFsm
>     raise origExc.with_traceback(origTraceback)
>   File "/app/.venv/lib/python3.10/site-packages/pysnmp/smi/instrum.py", line 221, in flipFlopFsm
>     rval = f(tuple(name), val, idx, acInfo)
>   File "/app/.venv/lib/python3.10/site-packages/pysnmp/smi/mibs/SNMPv2-SMI.py", line 532, in writeTest
>     node.writeTest(name, val, idx, acInfo)
>   File "/app/.venv/lib/python3.10/site-packages/pysnmp/smi/mibs/SNMPv2-SMI.py", line 1211, in writeTest
>     self.__delegate('Test', name, val, idx, acInfo)
>   File "/app/.venv/lib/python3.10/site-packages/pysnmp/smi/mibs/SNMPv2-SMI.py", line 1181, in __delegate
>     getattr(self.getBranch(name, idx), 'write' + subAction)(
>   File "/app/.venv/lib/python3.10/site-packages/pysnmp/smi/mibs/SNMPv2-SMI.py", line 957, in writeTest
>     MibScalar.writeTest(self, name, val, idx, acInfo)
>   File "/app/.venv/lib/python3.10/site-packages/pysnmp/smi/mibs/SNMPv2-SMI.py", line 633, in writeTest
>     MibTree.writeTest(self, name, val, idx, acInfo)
>   File "/app/.venv/lib/python3.10/site-packages/pysnmp/smi/mibs/SNMPv2-SMI.py", line 532, in writeTest
>     node.writeTest(name, val, idx, acInfo)
>   File "/app/.venv/lib/python3.10/site-packages/pysnmp/smi/mibs/SNMPv2-SMI.py", line 740, in writeTest
>     raise error.WrongValueError(idx=idx, name=name, msg=sys.exc_info()[1])
> pysnmp.smi.error.WrongValueError: WrongValueError({'idx': 3, 'name': (1, 3, 6, 1, 6, 3, 15, 1, 2, 2, 1, 5, 5, 128, 0, 58, 140, 4, 6, 85, 83, 69, 82, 48, 50), 'msg': WrongValueError()})
> [scsnmp@wk2 sc4snmp]$ 
>

**Comment by isrijaya** [comment]
Update: using SHA256 works!

microk8s kubectl create -n sc4snmp secret generic snmpv3-secret-sha2-aes \
  --from-literal=userName=USER02 \
  --from-literal=authKey=xxxxxx \
  --from-literal=privKey=xxxxxxx \
  --from-literal=authProtocol=SHA256\
  --from-literal=privProtocol=AES256

**Reply by ajasnosz** [reply]
Great to hear that it's working!
